<a href="https://colab.research.google.com/github/alisial94/DV3_DataVisualisation_UnicornCompanies/blob/main/DV3_FinalProject_AliSial.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Unicorn Companies - Data Visualisation 
__________________________________________

### Data Visualisation 3 - Final Project
__________________________________________

## Data and Analytical Questions

The dataset is obtained form *__[Mavren Analytics](https://www.mavenanalytics.io/data-playground)__*, there is a variety of really good datasets that you can work with and upscale your data analytics skills. I downloaded the dataset and saved it into my personal GitHub repository. You can directly access the data and colab notebook by clicking *__[HERE](https://github.com/alisial94/DV3_DataVisualisation_UnicornCompanies)__*. 

Unicorn Dataset, the one I am working with, comprises of 1074 private companies with a valuation over $1 billion as of March 2022, including each company's current valuation, funding, country of origin, industry, select investors, and the years they were founded and became unicorns. It would be interesting to see how much have these companies grown in valuation compared to the funding they recieved to start. Below you can find a list of analytical questions I would like to answer as part of the analysis and these questions are provided by Mavern Analytics. 

- Which unicorn companies have had the biggest return on investment?
- How long does it usually take for a company to become a unicorn? Has it always been this way?
- Which countries have the most unicorns? Are there any cities that appear to be industry hubs?
- Which investors have funded the most unicorns?

now let's jump into the analysis and answer these questions. 


 ## Data Cleaning & Exploratory Analysis

In this section we will be exploring the data to see how it looks and if it requires any transformations before we proceed to analysising the questions. After obtaining the final dataset, I'll also perform some basic EDA. 

### Importing required libraries

In [ ]:
# importing libraries
import pandas as pd
import plotly.express as px
import numpy as np

# install dash app
! pip install jupyter-dash # For Dash applications


# import the Dash core and HTML components
from dash import dcc
from dash import html

# we need to import the input and output functions for the interactive features
from dash import Input, Output


# import the JupyterDash extension
from jupyter_dash import JupyterDash

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 9.8 MB 14.2 MB/s 
     |████████████████████████████████| 357 kB 38.8 MB/s 
  Created wheel for retrying: filename=retrying-1.3.3-py3-none-any.whl size=11447 sha256=8ab7e7fd6a69599c01af165dd3a61a167ea3b3a6ce4ba5987a7e0ffb9c6018da
  Stored in directory: /root/.cache/pip/wheels/f9/8d/8d/f6af3f7f9eea3553bc2fe6d53e4b287dad18b06a861ac56ddf
Successfully built retrying


In [ ]:
#I had to do this as the visualizations were not showing up on the screen
#def configure_plotly_browser_state():
#  import IPython
#  display(IPython.core.display.HTML('''
#        <script src="/static/components/requirejs/require.js"></script>
#        <script>
#          requirejs.config({
#            paths: {
#              base: '/static/base',
#              plotly: 'https://cdn.plot.ly/plotly-latest.min.js?noext',
#            },
#          });
#        </script>
#        '''))

### Loding the Data

In [ ]:
# loading the data from the git repo

df = pd.read_csv('https://raw.githubusercontent.com/alisial94/DV3_DataVisualisation_UnicornCompanies/main/Data/Unicorn_Companies.csv')
df.head()

Company Valuation Date Joined                         Industry  \
0  Bytedance     $180B  2017-04-07          Artificial intelligence   
1     SpaceX     $100B  2012-12-01                            Other   
2      SHEIN     $100B  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe      $95B  2014-01-23                          Fintech   
4     Klarna      $46B  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded Funding  \
0        Beijing          China           Asia          2012     $8B   
1      Hawthorne  United States  North America          2002     $7B   
2       Shenzhen          China           Asia          2008     $2B   
3  San Francisco  United States  North America          2010     $2B   
4      Stockholm         Sweden         Europe          2005     $4B   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Founders Fund, Draper Fisher Jurvetson, Rothen...  
2  Tiger Global Management, Sequoia Capital China...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4  Institutional Venture Partners, Sequoia Capita...

In [ ]:
# checking the data type of each column to fix it according to our requirement

df.dtypes

Company             object
Valuation           object
Date Joined         object
Industry            object
City                object
Country             object
Continent           object
Year Founded         int64
Funding             object
Select Investors    object
dtype: object

### Transforming the Data 

It is clear that most of our columns are strings including the valuation and funding columns. To proceed any further we need the columns in integer format. There are few companies whose funding is unknown, I'll be removing them from the dataset for obvious reasons. Next I will be adjusting the date column to datatime format and changing the year column to integer. I'll also be spliting the investor column into four new columns

#### 1 - Valuation Column

In [ ]:
# fixing the column type of valuation

# replacing strings
df['Valuation'] = df['Valuation'].str.replace("$","")
df['Valuation'] = df['Valuation'].str.replace("B","000000000")

# changing type to int
df['Valuation'] = df['Valuation'].astype(int)

df.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  after removing the cwd from sys.path.


Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded Funding  \
0        Beijing          China           Asia          2012     $8B   
1      Hawthorne  United States  North America          2002     $7B   
2       Shenzhen          China           Asia          2008     $2B   
3  San Francisco  United States  North America          2010     $2B   
4      Stockholm         Sweden         Europe          2005     $4B   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Founders Fund, Draper Fisher Jurvetson, Rothen...  
2  Tiger Global Management, Sequoia Capital China...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4  Institutional Venture Partners, Sequoia Capita...

#### 2 - Funding 

In [ ]:
# fixing the column type of Funding

# replacing strings
df = df[df['Funding'] != 'Unknown']
df = df[df['Funding'] != '0']
df['Funding'] = df['Funding'].str.replace("$","")
df['Funding'] = df['Funding'].str.replace("B","000000000")
df['Funding'] = df['Funding'].str.replace("M","000000")


# changing type to int
df['Funding'] = df['Funding'].astype(int)

df.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:6: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  


Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Founders Fund, Draper Fisher Jurvetson, Rothen...  
2  Tiger Global Management, Sequoia Capital China...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4  Institutional Venture Partners, Sequoia Capita...

In [ ]:
# one company has a value of zero, and was formed in 1919, i'll be removing this as well.
df[df['Company'] == 'Otto Bock HealthCare']

Company   Valuation Date Joined Industry        City  \
189  Otto Bock HealthCare  4000000000  2017-06-24   Health  Duderstadt   

     Country Continent  Year Founded  Funding Select Investors  
189  Germany    Europe          1919        0     EQT Partners

In [ ]:
df = df[df['Company'] != 'Otto Bock HealthCare']

#### 3 - Date Joined

I'll also be extracting the year from date joined column which will be used later on in the analysis.

In [ ]:
# transforming Date Joined column

df['Date Joined'] = pd.to_datetime(df['Date Joined'], format='%Y-%m-%d')
df['Year Joined'] = pd.DatetimeIndex(df['Date Joined']).year



df.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.


Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  
0  Sequoia Capital China, SIG Asia Investments, S...         2017  
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012  
2  Tiger Global Management, Sequoia Capital China...         2018  
3        Khosla Ventures, LowercaseCapital, capitalG         2014  
4  Institutional Venture Partners, Sequoia Capita...         2011

#### 4 - Investors

While exploring the investor column, I realised that at max there are 4 investors listed against one company. Therefore, I decided to split the investor column into 4 sub columns. 

In [ ]:
# splitiing investors into 4 columns

df[['First Investor','Second Investor', 'Third Investor','Fourth Investor']] = df['Select Investors'].str.split(',', n=3, expand=True)


# removing any leading and trailing space
df['First Investor'] = df['First Investor'].str.strip()
df['Second Investor'] = df['Second Investor'].str.strip()
df['Third Investor'] = df['Third Investor'].str.strip()
df['Fourth Investor'] = df['Fourth Investor'].str.strip()
df.head()



/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py:3641: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[k1] = value[k2]


Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor  
0                Sina Weibo  Softbank Group  
1       Rothenberg Ventures            None  
2  Shunwei Capital Partners            None  
3                  capitalG            None  
4          General Atlantic            None

### Final Dataset

As you just saw above, our final dataset contains a total of 15 columns apart from the index and at this point I will not be dropping any columns. Now lets look at out some basic visualistaions.

In [ ]:
df.head()

Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor  
0                Sina Weibo  Softbank Group  
1       Rothenberg Ventures            None  
2  Shunwei Capital Partners            None  
3                  capitalG            None  
4          General Atlantic            None

### Top 10 companies based on valuation

In [ ]:

#configure_plotly_browser_state()
#init_notebook_mode(connected = False)


# Looking at the arranging the dataset

df_top10_val = df.sort_values('Valuation', axis = 0, ascending = False).head(10)

# creating the lable 
labels_1 = {'Company': 'Company Name', 'Valuation':"Total Valuation"}

# ploting the graph
Top_10_hist = px.histogram(df_top10_val, x='Company', y='Valuation',
                      color = 'Company', 
                      #opacity=0.75,
                      color_discrete_map={"Bytedance" : '#001219', "SpaceX": '#005f73', "SHEIN": '#0a9396', 
                                          "Stripe" : '#94d2bd', "Klarna" : '#e9d8a6', "Canva": '#ee9b00',
                                          "Checkout.com":'#ca6702', "Instacart": '#bb3e03',
                                          "JUUL Labs": '#ae2012', "Databricks": '#9b2226'},
                      template='simple_white', 
                      title = "Top 10 Companies by Valuation",
                      labels = labels_1,
                      hover_name= 'Company', 
                      hover_data={'Company' : False,
                                  'Year Founded' : True,
                                  'Funding' : True,
                                  'Valuation' : False})

Top_10_hist.update_layout( yaxis_title="Total Valuation", hovermode="x unified")
                          

Top_10_hist.show()

## Question 1 - Which unicorn companies have had the biggest return on investment?

For this question we will be calculating a new field in our dataset which will be named ROI (Retuen on Investment). This field will be calculated using funding amount and valuation amount.

### Adding the new calculated field 

In [ ]:
# calculating ROI
df['ROI'] = (df['Valuation'] - df['Funding'])/df['Funding']

# changing it to two decimal places
df = df.round(2)
df.head()

Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor    ROI  
0                Sina Weibo  Softbank Group  21.50  
1       Rothenberg Ventures            None  13.29  
2  Shunwei Capital Partners            None  49.00  
3                  capitalG            None  46.50  
4          General Atlantic            None  10.50

### Ploting the Top 10 Companies in terms of ROI

In [ ]:
#configure_plotly_browser_state()
#init_notebook_mode(connected = False)


# Looking at the arranging the dataset

df_top10_roi = df.sort_values('ROI', axis = 0, ascending = False).head(10)
#df_top10_roi_noZap = df_top10_roi_noZap[df_top10_roi_noZap['Company'] != 'Zapier']

# creating the lable 
labels_2 = {'Company':'Company Name', 'ROI':"Return on Investment"}

# ploting the graph
Top_10_roi_hist = px.histogram(df_top10_roi, x='Company', y='ROI',
                      color = 'Company',
                      #opacity=0.75, 
                      color_discrete_map={"Zapier" : '#001219', "Dunamu": '#005f73', "Workhuman": '#0a9396', 
                                          "CFGI" : '#94d2bd', "Manner" : '#e9d8a6', "DJI Innovation": '#ee9b00',
                                          "GalaxySpace":'#ca6702', "Canva": '#bb3e03',
                                          "Il Makiage": '#ae2012', "Revolution Precrafted": '#9b2226'},
                      template='simple_white', 
                      title = "Top 10 Companies by Return on Investment (without Zapier)",
                      labels = labels_2,
                      hover_name= 'Company', 
                      hover_data={'Company' : False})

Top_10_roi_hist.update_layout( yaxis_title="Total Return on Investment", hovermode="x unified")
                          

Top_10_roi_hist.show()

Based on the graph above we can definitely say that Zapier appears to have the highest return on investment from the list of companies remaining in our dataset. Including Zapier in the top 10 has resulted in a poor performing graph as we cannot visualise the difference between the ROIs of the other nine companies. In the next graph you will see Top 10 companies excluding Zapier. 

In [ ]:
#configure_plotly_browser_state()
#init_notebook_mode(connected = False)


# Looking at the arranging the dataset

df_top10_roi_noZap = df.sort_values('ROI', axis = 0, ascending = False).head(11)
df_top10_roi_noZap = df_top10_roi_noZap[df_top10_roi_noZap['Company'] != 'Zapier']

# creating the lable 
labels_2 = {'Company':'Company Name', 'ROI':"Return on Investment"}

# ploting the graph
Top_10_roi_hist = px.histogram(df_top10_roi_noZap, x='Company', y='ROI',
                      color = 'Company',
                      #opacity=0.75, 
                      color_discrete_map={"Dunamu" : '#001219', "Workhuman": '#005f73', "CFGI": '#0a9396', 
                                          "Manner" : '#94d2bd', "DJI Innovation" : '#e9d8a6', "GalaxySpace": '#ee9b00',
                                          "Canva":'#ca6702', "Il Makiage": '#bb3e03',
                                          "Revolution Precrafted": '#ae2012', "Injective Protocol": '#9b2226'},
                      template='simple_white', 
                      title = "Top 10 Companies by Return on Investment (without Zapier)",
                      labels = labels_2,
                      hover_name= 'Company', 
                      hover_data={'Company' : False})

Top_10_roi_hist.update_layout( yaxis_title="Total Return on Investment", hovermode="x unified")
                          

Top_10_roi_hist.show()

## Question 2 - How long does it usually take for a company to become a unicorn? Has it always been this way?

This is a tricky question I would say, because the approach used to obtain the final result will have a very subjective route. Why is that? simply becuase you will have to make some arbitrary assumptions to calculate the time it took for a company to grow and reach $1 Billion in valuation. 

The first thing I explored was the distinct funding values below $500 Million to capture the impact of growth. This value could have been something else based on what you assume would be a good valuation to begin with. Next I looked up the difference between the year each of these company were founded and also the year the reach 1 Billion in valuation. Below you will see the code I used to achieve this.   

### Caluation Difference in year

In [ ]:
# filter the data to $500 Million starting capital

df_fund500 = df[df['Funding'] != 500000000]
df_fund500.head()

Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor    ROI  
0                Sina Weibo  Softbank Group  21.50  
1       Rothenberg Ventures            None  13.29  
2  Shunwei Capital Partners            None  49.00  
3                  capitalG            None  46.50  
4          General Atlantic            None  10.50

In [ ]:
# adding the calculated field for year to grow to 1 Billion

df_fund500['Growth Year'] = df_fund500['Year Joined'] - df_fund500['Year Founded']
df_fund500.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor    ROI  Growth Year  
0                Sina Weibo  Softbank Group  21.50            5  
1       Rothenberg Ventures            None  13.29           10  
2  Shunwei Capital Partners            None  49.00           10  
3                  capitalG            None  46.50            4  
4          General Atlantic            None  10.50            6

Now we have the number of years it took for these companies to enter the category of unicorn. Next i'll be taking an average of this new calculated field to find out how much time on average it would take for any company to grow and become a unicorn (reach 1 Billion). The calculation is performed below but the value comes up to be around 7 years. This value, as I said earlier, is arbitrary and resulted due assumptions I made for these calculations.

In [ ]:
# Average number of years it takes a company to become Unicorn

df_fund500['Growth Year'].mean()


6.906515580736544

### Second part of the this question

Even though we have an answer for the average number of years it takes a company to become a unicorn, the second part of the question asks us to include time as a categorical variable. What I mean by that is we use year founded data column and split it into 3 decades and then calculate the mean number of years it took for a company formed within a decade to become a unicorn. Later I compared the mean number of years amoung the decades to see if the value we calculated earlier changes over the decades or whatever timeframe you wish to divide the data in. I'll be dividing the data as follow:

- 1990-1999
- 2000-2009
- 2010-2020

In [ ]:
# Dividing the data

# 1990-1999
df_90 = df_fund500[(df_fund500['Year Founded'] >=1990) & (df_fund500['Year Founded'] < 2000)] 

# 2000-2009
df_00 = df_fund500[(df_fund500['Year Founded'] >=2000) & (df_fund500['Year Founded'] < 2010)] 

# 2010-2019
df_10 = df_fund500[(df_fund500['Year Founded'] >=2010) & (df_fund500['Year Founded'] < 2020)] 


In [ ]:
# average years for companies formed in 90s

avg_90 = df_90['Growth Year'].mean()
avg_90

21.391304347826086

In [ ]:
# average years for companies formed in 2000s

avg_00 = df_00['Growth Year'].mean()
avg_00

12.771812080536913

In [ ]:
# average years for companies formed in 2010 onwards

avg_10 = df_10['Growth Year'].mean()
avg_10


5.669411764705882

In [ ]:
# creating a new dataframe for visualisations

# initialize list of lists
data = [['1990-99', avg_90], ['2000-10', avg_00], ['2010-20', avg_10]]

# Create the pandas DataFrame
decade_avg = pd.DataFrame(data, columns=['Decade', 'Avg Years']).round(2)
decade_avg


Decade  Avg Years
0  1990-99      21.39
1  2000-10      12.77
2  2010-20       5.67

In [ ]:
#configure_plotly_browser_state()
#init_notebook_mode(connected = False)


# ploting the avg year graph 

# setting colours
#dec_col = ("1990-99" = '#3D9970', '2000-10'= '#FF4136', '2010-20'= '#FF851B')

decade_avg_hist = px.histogram(decade_avg, x='Decade', y='Avg Years',
                               color='Decade', 
                               opacity=0.80,
                               color_discrete_map={"1990-99" : '#3D9970', "2000-10": '#FF4136', "2010-20": '#FF851B'}, 
                               template='simple_white', 
                               title = "Avg years by Deacades",
                               hover_name= 'Decade', 
                               hover_data={'Decade' : False})

decade_avg_hist.update_layout( yaxis_title="Average Years", hovermode="x unified")
                          

decade_avg_hist.show()




# Box plot for overall distribtion of number of years by the decades
import plotly.graph_objects as go

x1 = ['1990-99', '1990-99', '1990-99', '1990-99', '1990-99', '1990-99',
     '1990-99','1990-99','1990-99', '1990-99','1990-99', '1990-99','1990-99',
      '1990-99','1990-99','1990-99','1990-99','1990-99','1990-99','1990-99','1990-99',
      '1990-99',]



dec_avg = go.Figure()
dec_avg.add_trace(go.Box(x = x1, y=df_90['Growth Year'],
                     name='1990-99',
                     marker_color='#3D9970'))
dec_avg.add_trace(go.Box(y=df_00['Growth Year'],
                     name='2000-10',
                     marker_color='#FF4136'))
dec_avg.add_trace(go.Box(y=df_10['Growth Year'],
                     name='2010-20',
                     marker_color='#FF851B'))

dec_avg.update_layout( yaxis_title="Average Years", 
                  xaxis_title = "Decades", 
                  title = "Distribution of Number of Years to reach Unicorn threshold",
                  template='simple_white')

dec_avg.show()

Based on the graphs above we can see that the average number of years it took for a company to be recognised as a Unicorn has reduced over the 3 decades. This clearly indicates that we now live in a more fast paced global enviornment where companies have a better opportunity to be successful and grow.  

## Question 3 - Which countries have the most unicorns? Are there any cities that appear to be industry hubs?



In this section we will look at the number of Unicorn Companies in each country and also try to analyse if there are any cities that might appear to be industry hubs. I will be creating a map to highlight the countries that have at least one Unicorn Company. In order to do so I have included another dataset that includes latitude and longitude data for the countries, I will be importing this data directly from the project repository at GitHub.

### Number of Unicorn per country

In [ ]:
# taking a count of unicorn companies in each country and save them in a new dataset

df_country = df.groupby(['Country']).size().reset_index(name='Count of Unicorn').sort_values('Count of Unicorn',ascending = False)
df_country.head()

Country  Count of Unicorn
44   United States               555
9            China               170
19           India                65
43  United Kingdom                43
17         Germany                25

In [ ]:
# importing lat long dataset

lat_long_df = pd.read_csv('https://raw.githubusercontent.com/alisial94/DV3_DataVisualisation_UnicornCompanies/main/Data/countries_data.csv')
lat_long_df.head(10)

country_code   latitude  longitude               country
0           AD  42.546245   1.601554               Andorra
1           AE  23.424076  53.847818  United Arab Emirates
2           AF  33.939110  67.709953           Afghanistan
3           AG  17.060816 -61.796428   Antigua and Barbuda
4           AI  18.220554 -63.068615              Anguilla
5           AL  41.153332  20.168331               Albania
6           AM  40.069099  45.038189               Armenia
7           AN  12.226079 -69.060087  Netherlands Antilles
8           AO -11.202692  17.873887                Angola
9           AQ -75.250973  -0.071389            Antarctica

In [ ]:
# renaming the columns

lat_long_df.rename(columns = {'country_code':'Country Code', 'latitude':'Latitude', 'longitude':'Longitude', 'country':'Country'}, inplace = True)
lat_long_df.head(2)

Country Code   Latitude  Longitude               Country
0           AD  42.546245   1.601554               Andorra
1           AE  23.424076  53.847818  United Arab Emirates

In [ ]:
# merging the two columns

df_country = pd.merge(df_country,lat_long_df,on='Country')
df_country.head()

Country  Count of Unicorn Country Code   Latitude   Longitude
0   United States               555           US  37.090240  -95.712891
1           China               170           CN  35.861660  104.195397
2           India                65           IN  20.593684   78.962880
3  United Kingdom                43           GB  55.378051   -3.435973
4         Germany                25           DE  51.165691   10.451526

In [ ]:
#configure_plotly_browser_state()
#init_notebook_mode(connected = False)



# creating the map

map = px.scatter_geo(df_country , lat = 'Latitude' , lon = 'Longitude',
                     color="Count of Unicorn", # which column to use to set the color of markers
                     hover_name="Country", # column added to hover information
                     size="Count of Unicorn", # size of markers
                     projection="natural earth")

map.update_layout(
        title = 'Countries by Number of Unicorn companies<br>(Hover for county names)',
        geo = dict(
            scope = 'world',
            landcolor = 'rgb(217, 217, 217)'))

map.show()

Observing the map above it is very obvious that USA has the most number of Unicorn Countries, having a count of 555 followed by China and India. It would be more interesting to see how the map looks like if we remove these top 3 countries from the equation. In the next section we will do that and try to highlight other emerging markets in the world. 

In [ ]:
#configure_plotly_browser_state()
#init_notebook_mode(connected = False)


# creating the map without USA, China or India

# filter the data
df_country_notop3 = df_country[df_country['Country Code'] != 'US']
df_country_notop3 = df_country_notop3[df_country['Country Code'] != 'CN']
df_country_notop3 = df_country_notop3[df_country['Country Code'] != 'IN']

# creat map
map_1 = px.scatter_geo(df_country_notop3 , lat = 'Latitude' , lon = 'Longitude',
                     color="Count of Unicorn", # which column to use to set the color of markers
                     hover_name="Country", # column added to hover information
                     size="Count of Unicorn", # size of markers
                     projection="natural earth")

map_1.update_layout(
        title = 'Countries by Number of Unicorn companies with Top 3<br>(Hover for county names)',
        geo = dict(
            scope = 'world',
            landcolor = 'rgb(217, 217, 217)'))

map_1.show()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:9: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:10: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.



### Cities vs Industry Hub

In this section, I plan to analyse if there are any cities that might become hubs for a particular sector of the industry. The idea here is to group the total industry count by city and then see if we have any winners. For this project I will be assuming that an industry hub is considered a city having 8 or more companies from the same industry.

In [ ]:
# grouping the data by city and industy

df_city = df.groupby(['City', 'Industry']).size().reset_index(name='Count of Industry').sort_values('Count of Industry',ascending = False)
df_city = df_city.head(16)
df_city

City                         Industry  Count of Industry
403  San Francisco     Internet software & services                 54
400  San Francisco                          Fintech                 41
318       New York                          Fintech                 33
243         London                          Fintech                 24
321       New York     Internet software & services                 19
320       New York                           Health                 14
402  San Francisco                           Health                 12
29         Beijing  E-commerce & direct-to-consumer                 11
453       Shanghai            Auto & transportation                 10
48       Bengaluru     Internet software & services                  9
35         Beijing      Mobile & telecommunications                  9
314       New York                    Cybersecurity                  9
396  San Francisco                    Cybersecurity                  8
30         Beijing                           Edtech                  8
394  San Francisco          Artificial intelligence                  8
26         Beijing          Artificial intelligence                  8

In [ ]:
#configure_plotly_browser_state()
#init_notebook_mode(connected = False)



# tree map for city wide industy hubs

tree_map = px.treemap(df_city, path=["City", 'Industry'], values='Count of Industry',
                  color='Count of Industry',
                  color_continuous_scale='Viridis',
                  hover_data= {'City':False,
                               'Industry':False}
                 )
tree_map.update_layout(margin = dict(t=50, l=25, r=25, b=25))
tree_map.show()

Based on the the treemap above we can say that we do see some cities as industry hubs, for example, San Francisco appears to be industry for Internet Software Services and also Fintech.   

## Question 4 - Which investors have funded the most unicorns?

If you remember that during the EDA phase, I decided to split the investor column in four new columns, that was mainly done to answer this question. Now we will be uing those columns to find the investors with most number of investments.

In [ ]:
df_first_investor = df.groupby(['First Investor']).size().reset_index(name='Count of First').sort_values('Count of First',ascending = False)
df_first_investor.rename(columns = {'First Investor':'Investor'}, inplace = True)
#df_first_investor['Investor'] = df_first_investor['Investor'].astype(str)
df_first_investor.head(10)

Investor  Count of First
420        Sequoia Capital China              30
37           Andreessen Horowitz              29
11                         Accel              28
252             Insight Partners              22
334    New Enterprise Associates              16
419              Sequoia Capital              16
245               Index Ventures              14
484      Tiger Global Management              14
297  Lightspeed Venture Partners              13
441               SoftBank Group              13

In [ ]:
df_second_investor = df.groupby(['Second Investor']).size().reset_index(name='Count of Second').sort_values('Count of Second',ascending = False)
df_second_investor.rename(columns = {'Second Investor':'Investor'}, inplace = True)
df_second_investor.head()

Investor  Count of Second
531  Tiger Global Management               20
17                     Accel               18
466          Sequoia Capital               17
227         General Catalyst               16
289         Insight Partners               14

In [ ]:
df_third_investor = df.groupby(['Third Investor']).size().reset_index(name='Count of Third').sort_values('Count of Third',ascending = False)
df_third_investor.rename(columns = {'Third Investor':'Investor'}, inplace = True)
df_third_investor.head()

Investor  Count of Third
505      Tiger Global Management              19
432              Sequoia Capital              13
14                         Accel              13
35           Andreessen Horowitz              12
307  Lightspeed Venture Partners              11

In [ ]:
df_fourth_investor = df.groupby(['Fourth Investor']).size().reset_index(name='Count of Fourth').sort_values('Count of Fourth',ascending = False)
df_fourth_investor.rename(columns = {'Fourth Investor':'Investor'}, inplace = True)
df_fourth_investor.head(10)

Investor  Count of Fourth
0                           Accel                1
1                   Goldman Sachs                1
2               McKesson Ventures                1
3               Redpoint Ventures                1
4             Redpoint e.ventures                1
5  SDIC CMC Investment Management                1
6                  Softbank Group                1
7                   Spark Capital                1

In [ ]:
# merging the tables for final table

df_investors = pd.merge(df_first_investor,df_second_investor, how='left')
df_investors = df_investors.merge(df_third_investor, on='Investor', how='left')
df_investors = df_investors.merge(df_fourth_investor, on='Investor', how='left')
#df_investors = df_investors.set_index('Investor')
df_investors.head()

Investor  Count of First  Count of Second  Count of Third  \
0      Sequoia Capital China              30              7.0            10.0   
1        Andreessen Horowitz              29             12.0            12.0   
2                      Accel              28             18.0            13.0   
3           Insight Partners              22             14.0            10.0   
4  New Enterprise Associates              16              4.0             3.0   

   Count of Fourth  
0              NaN  
1              NaN  
2              1.0  
3              NaN  
4              NaN

In [ ]:
# removing NaN values and fixing the type of columns

df_investors['Count of Second'] = df_investors['Count of Second'].fillna(0).astype(int)
df_investors['Count of Third'] = df_investors['Count of Third'].fillna(0).astype(int)
df_investors['Count of Fourth'] = df_investors['Count of Fourth'].fillna(0).astype(int)
df_investors.head()

Investor  Count of First  Count of Second  Count of Third  \
0      Sequoia Capital China              30                7              10   
1        Andreessen Horowitz              29               12              12   
2                      Accel              28               18              13   
3           Insight Partners              22               14              10   
4  New Enterprise Associates              16                4               3   

   Count of Fourth  
0                0  
1                0  
2                1  
3                0  
4                0

In [ ]:
# calculating the top number of investments

df_investors['Total Investments'] = df_investors['Count of First'] + df_investors['Count of Second'] + df_investors['Count of Third'] + df_investors['Count of Fourth']
df_investors = df_investors.sort_values('Total Investments',ascending = False)
df_investors.head(15)

Investor  Count of First  Count of Second  \
2                         Accel              28               18   
7       Tiger Global Management              14               20   
1           Andreessen Horowitz              29               12   
0         Sequoia Capital China              30                7   
3              Insight Partners              22               14   
5               Sequoia Capital              16               17   
14             General Catalyst              10               16   
8   Lightspeed Venture Partners              13               10   
9                SoftBank Group              13               12   
6                Index Ventures              14                9   
10             Tencent Holdings              12                8   
15             General Atlantic               9                8   
28              Google Ventures               6               11   
12        Sequoia Capital India              11                8   
27                  IDG Capital               6                8   

    Count of Third  Count of Fourth  Total Investments  
2               13                1                 60  
7               19                0                 53  
1               12                0                 53  
0               10                0                 47  
3               10                0                 46  
5               13                0                 46  
14               8                0                 34  
8               11                0                 34  
9                9                0                 34  
6                9                0                 32  
10               9                0                 29  
15              11                0                 28  
28               8                0                 25  
12               6                0                 25  
27               9                0                 23

In [ ]:
#configure_plotly_browser_state()
#init_notebook_mode(connected = False)



# plotting the top 15 investor results

bubble_chart = px.scatter(df_investors.head(15), x='Investor', y="Total Investments",
	         size="Total Investments", color="Total Investments",
                 hover_name="Investor", size_max=60,
                 hover_data={'Investor' : False})
bubble_chart.update_layout( title = "Top 15 investors based on Investments",
                  template='simple_white')
bubble_chart.show()

## Question 5 - Valuation vs Funding vs ROI

For this question we will be looking at relationship between Valuation, Funding and ROI. We will be making two graphs, one with top 10 and one without top 10. Ine the second graph we will be removing Zapier as well.

### 3D Scatter Plot - Valuation vs Funding vs ROI

In [ ]:
# All compnaies

#configure_plotly_browser_state()
#init_notebook_mode(connected = False)


#df_150 = df.sort_values(by='Valuation',ascending=False)
#df_150=df_150.iloc[:150,:]

fig = px.scatter_3d(df, x='Funding', y='ROI', z='Valuation',
              color='Valuation', size='Valuation', size_max=25,
              hover_name="Company", 
              opacity=0.8)

# tight layout
fig.update_layout(title = "Valuation vs Funding Vs ROI - All Companies",
                  margin=dict(l=0, r=0, b=0, t=0)
                  )




In [ ]:

# Compnaies without Top 10 and Zapier

#configure_plotly_browser_state()
#init_notebook_mode(connected = False)

# Adjusting the data for better ploting
df_new = df.sort_values(by='Valuation',ascending=False)
df_new=df_new.iloc[10:]
df_new = df_new[df_new['Company'] != 'Zapier']


fig = px.scatter_3d(df_new, x='Funding', y='ROI', z='Valuation',
                    color='Valuation', size='Valuation', size_max=25,
                    hover_name="Company", opacity=0.8)

# tight layout
fig.update_layout(title = "Valuation vs Funding Vs ROI",
                  margin=dict(l=0, r=0, b=0, t=0)
                  )

Based on the graph above we can see that the higher the funding the higher valuation, which kind of makes sense. But what's more interesting is that these companies tend to have a lower ROI compared to companies with less funding or valuation. 

## Question 6 - Dash App with Top 10 companies in a Industry

In [ ]:
df['Industry'].unique()

array(['Artificial intelligence', 'Other',
       'E-commerce & direct-to-consumer', 'Fintech',
       'Internet software & services',
       'Supply chain, logistics, & delivery', 'Consumer & retail',
       'Data management & analytics', 'Edtech', 'Health', 'Hardware',
       'Auto & transportation', 'Travel', 'Cybersecurity',
       'Mobile & telecommunications', 'Artificial Intelligence'],
      dtype=object)

In [ ]:
df['Industry'].replace('Artificial intelligence', 'Artificial Intelligence', inplace=True)

In [ ]:
df_AI = df[df['Industry'] == 'Artificial Intelligence'].sort_values('Valuation',ascending = False).head(10)
df_ECOM = df[df['Industry'] == 'E-commerce & direct-to-consumer'].sort_values('Valuation',ascending = False).head(10)
df_Fin = df[df['Industry'] == 'Fintech'].sort_values('Valuation',ascending = False).head(10)
df_IT = df[df['Industry'] == 'Internet software & services'].sort_values('Valuation',ascending = False).head(10)
df_SC = df[df['Industry'] == 'Supply chain, logistics, & delivery'].sort_values('Valuation',ascending = False).head(10)
df_Retail = df[df['Industry'] == 'Consumer & retail'].sort_values('Valuation',ascending = False).head(10)
df_DataMan = df[df['Industry'] == 'Data management & analytics'].sort_values('Valuation',ascending = False).head(10)
df_Edtech = df[df['Industry'] == 'Edtech'].sort_values('Valuation',ascending = False).head(10)
df_Health = df[df['Industry'] == 'Health'].sort_values('Valuation',ascending = False).head(10)
df_Hardware = df[df['Industry'] == 'Hardware'].sort_values('Valuation',ascending = False).head(10)
df_Auto = df[df['Industry'] == 'Auto & transportation'].sort_values('Valuation',ascending = False).head(10)
df_Travel = df[df['Industry'] == 'Travel'].sort_values('Valuation',ascending = False).head(10)
df_CyberSec = df[df['Industry'] == 'Cybersecurity'].sort_values('Valuation',ascending = False).head(10)
df_MobTel = df[df['Industry'] == 'Mobile & telecommunications'].sort_values('Valuation',ascending = False).head(10)

df_industry = pd.concat([df_AI,df_ECOM,df_Fin,df_IT,df_SC,df_Retail,df_DataMan,df_Edtech,df_Health,df_Hardware,df_Auto,df_Travel,df_CyberSec,df_MobTel])


df_industry

Company     Valuation Date Joined                     Industry  \
0       Bytedance  180000000000  2017-04-07      Artificial Intelligence   
31          Faire   12000000000  2019-10-30      Artificial Intelligence   
29        Argo AI   12000000000  2019-07-12      Artificial Intelligence   
63        Pony.ai    9000000000  2018-07-11      Artificial Intelligence   
94       Scale AI    7000000000  2019-08-05      Artificial Intelligence   
..            ...           ...         ...                          ...   
172  Yello Mobile    4000000000  2014-11-11  Mobile & telecommunications   
222        Branch    4000000000  2018-09-10  Mobile & telecommunications   
226     Clubhouse    4000000000  2021-01-24  Mobile & telecommunications   
246         Yixia    3000000000  2015-11-24  Mobile & telecommunications   
252   MessageBird    3000000000  2020-10-08  Mobile & telecommunications   

              City        Country      Continent  Year Founded     Funding  \
0          Beijing          China           Asia          2012  8000000000   
31   San Francisco  United States  North America          2017  1000000000   
29      Pittsburgh  United States  North America          2016  4000000000   
63         Fremont  United States  North America          2016  1000000000   
94   San Francisco  United States  North America          2018   603000000   
..             ...            ...            ...           ...         ...   
172          Seoul    South Korea           Asia          2012   216000000   
222   Redwood City  United States  North America          2014   667000000   
226  San Francisco  United States  North America          2020   110000000   
246        Beijing          China           Asia          2011   775000000   
252      Amsterdam    Netherlands         Europe          2011  1000000000   

                                      Select Investors  Year Joined  \
0    Sequoia Capital China, SIG Asia Investments, S...         2017   
31   Khosla Ventures, Forerunner Ventures, Sequoia ...         2019   
29          Volkswagen Group, Ford Autonomous Vehicles         2019   
63    Sequoia Capital China, IDG Capital, DCM Ventures         2018   
94                 Accel, Y Combinator, Index Ventures         2019   
..                                                 ...          ...   
172                                        Formation 8         2014   
222   New Enterprise Associates, Pear, Cowboy Ventures         2018   
226                   Andreessen Horowitz, TQ Ventures         2021   
246  Sequoia Capital China, Sina Weibo, Kleiner Per...         2015   
252                       Y Combinator, Atomico, Accel         2020   

                First Investor           Second Investor  \
0        Sequoia Capital China      SIG Asia Investments   
31             Khosla Ventures       Forerunner Ventures   
29            Volkswagen Group  Ford Autonomous Vehicles   
63       Sequoia Capital China               IDG Capital   
94                       Accel              Y Combinator   
..                         ...                       ...   
172                Formation 8                      None   
222  New Enterprise Associates                      Pear   
226        Andreessen Horowitz               TQ Ventures   
246      Sequoia Capital China                Sina Weibo   
252               Y Combinator                   Atomico   

                       Third Investor    Fourth Investor    ROI  
0                          Sina Weibo     Softbank Group  21.50  
31                    Sequoia Capital               None  11.00  
29                               None               None   2.00  
63                       DCM Ventures               None   8.00  
94                     Index Ventures               None  10.61  
..                                ...                ...    ...  
172                              None               None  17.52  
222                   Cowboy Ventures               None  

In [ ]:
#configure_plotly_browser_state()
#init_notebook_mode(connected = False)



# create the Dash app object
app = JupyterDash(__name__)

# creating the app layout with a dropdown, and 2 charts
app.layout = html.Div(
[
    html.H1("Industy Top 10 Unicorn Companies", style = {'textAlign': 'center', 'color': 'White', 'font-family':'sans-serif'}),
    
    
    html.Div("Choose a Industry from the dropdown", 
             style = {'color': 'White',  'font-family':'sans-serif', 'margin-top':'5px'}),
 
    dcc.Dropdown( id= 'Industry', options=df_industry['Industry'].unique(), value=df_industry['Industry'][0], 
                 style = {'width': '35%', 'padding':'8px'} ),
 
    html.Div( dcc.Graph( id='graph1'))

]
)

@app.callback(
    Output('graph1', 'figure'),
    Input('Industry', 'value')
)
def update1( _input1 ):

  
  # Bar Chart for transaction volume
  df2 = df_industry.query('Industry == @_input1')
  bar_chart = px.histogram(df2, x = 'Company', y = 'Valuation',
             color = 'Company', 
             template='simple_white', 
                      labels = labels_1,
             hover_data={'Company' : False,
                         'Valuation' :':.0f'})
  bar_chart.update_traces(opacity = 0.8)
  bar_chart.update_layout(title= "Top 10 in " +  _input1 + " Industry")
  bar_chart.update_layout(showlegend=False)
                    
  return  bar_chart


app.run_server(mode='inline')

<IPython.core.display.Javascript object>